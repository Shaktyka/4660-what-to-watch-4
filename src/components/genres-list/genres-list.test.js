import React from 'react';
import renderer from 'react-test-renderer';

import GenresList from './genres-list.jsx';

const genres = [`All genres`, `Crime`, `Sci-Fi`, `Drama`, `Comedy`];

describe(`GenresList rendering`, () => {

  it(`GenresList renders correctly with the first active filter`, () => {
    const activeGenre = `All genres`;

    const tree = renderer
      .create(
          <GenresList
            genres={genres}
            activeGenre={activeGenre}
            onGenreClick={() => {}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`GenresList renders correctly with the third active filter`, () => {
    const activeGenre = `Sci-Fi`;

    const tree = renderer
      .create(
          <GenresList
            genres={genres}
            activeGenre={activeGenre}
            onGenreClick={() => {}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});

