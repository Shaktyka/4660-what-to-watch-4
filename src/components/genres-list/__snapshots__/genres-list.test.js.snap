// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresList rendering GenresList renders correctly with the first active filter 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Sci-Fi
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Comedy
    </a>
  </li>
</ul>
`;
